pipeline/src/handler.ts

```diff
@@ -20,34 +30,58 @@ export const createHandler =
       settings: articles.settings,
     });
 
-    // 1. Fetches everything from Prismic
+    // 1. Prepare queries to fetch everything from Prismic
     const window = toBoundedWindow(event);
+    log.info(`Fetching documents last published ${describeWindow(window)}`);
+
+    // 2. Fetch all documents published in the given window and partition them into
+    // "parent" and "other" documents: those which we want to transform and those
+    // which *may* have been denormalised onto the former.
+    const [parentDocuments, otherDocuments] = partition(
+      paginator((after?: string) =>
+        getPrismicDocuments(clients.prismic, {
+          publicationWindow: toBoundedWindow(event),
+          graphQuery,
+          after,
+        })
+      ),
+      isArticle
+    );
+
+    // 3. Index all the parent documents into elasticsearch after transforming them
+    // into our common schema
+    const initialIndex = await bulkIndexDocuments(
```

Comment:
Does the two-stage indexing thing do something important?

IIUIC, bulkIndexDocuments now throws when a single document fails, so if any of these top-level documents fails to index (possibly even for transient reasons), then none of the nextParentDocuments will be updated.

Could the indexing of all of them be done in one call?

Reply:
It did but I have now removed that - the last 3 commits should explain the situation. Re the throwing-on-failure, I decided that doing that (so we get a Slack alert) was the easiest approach for now!